vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/WalletService.kt

```diff
@@ -99,62 +210,120 @@ class WalletService(
             format = toFormat(),
-            types = arrayOf(VERIFIABLE_CREDENTIAL) + credentialScheme.vcType,
-            claims = requestedAttributes?.toRequestedClaims(),
+            credentialDefinition = SupportedCredentialFormatDefinition(
+                types = listOf(VERIFIABLE_CREDENTIAL, credentialScheme.vcType),
+            ),
+            claims = requestedAttributes?.toRequestedClaims(credentialScheme),
         )
 
         ConstantIndex.CredentialRepresentation.ISO_MDOC -> AuthorizationDetails(
             type = CREDENTIAL_TYPE_OPENID,
             format = toFormat(),
             docType = credentialScheme.isoDocType,
-            types = arrayOf(credentialScheme.vcType),
-            claims = requestedAttributes?.toRequestedClaims()
+            claims = requestedAttributes?.toRequestedClaims(credentialScheme)
         )
     }
 
-    private fun ConstantIndex.CredentialRepresentation.toCredentialRequestParameters(proof: CredentialRequestProof) =
-        when (this) {
-            ConstantIndex.CredentialRepresentation.PLAIN_JWT,
-            ConstantIndex.CredentialRepresentation.SD_JWT -> CredentialRequestParameters(
-                format = toFormat(),
-                claims = requestedAttributes?.toRequestedClaims(),
-                types = arrayOf(VERIFIABLE_CREDENTIAL) + credentialScheme.vcType,
-                proof = proof
-            )
-
-            ConstantIndex.CredentialRepresentation.ISO_MDOC -> CredentialRequestParameters(
-                format = toFormat(),
-                docType = credentialScheme.isoDocType,
-                claims = requestedAttributes?.toRequestedClaims(),
-                types = arrayOf(credentialScheme.vcType),
-                proof = proof
-            )
-        }
+    private fun ConstantIndex.CredentialRepresentation.toCredentialRequestParameters(
+        credentialScheme: ConstantIndex.CredentialScheme,
+        requestedAttributes: Collection<String>?,
+        proof: CredentialRequestProof
+    ) = when (this) {
+        ConstantIndex.CredentialRepresentation.PLAIN_JWT,
+        ConstantIndex.CredentialRepresentation.SD_JWT -> CredentialRequestParameters(
+            format = toFormat(),
+            claims = requestedAttributes?.toRequestedClaims(credentialScheme),
+            credentialDefinition = SupportedCredentialFormatDefinition(
+                types = listOf(VERIFIABLE_CREDENTIAL) + credentialScheme.vcType,
+            ),
+            proof = proof
+        )
+
+        ConstantIndex.CredentialRepresentation.ISO_MDOC -> CredentialRequestParameters(
+            format = toFormat(),
+            docType = credentialScheme.isoDocType,
+            claims = requestedAttributes?.toRequestedClaims(credentialScheme),
+            proof = proof
+        )
+    }
 
-    private fun Collection<String>.toRequestedClaims() =
+    private fun Collection<String>.toRequestedClaims(credentialScheme: ConstantIndex.CredentialScheme) =
```

Comment:
Maybe an issue to be solved in the future, it appears all over the place anyway (as it also has before):
Credentials may contain attributes from different namespaces

Reply:
So you mean we'll need to update our interface `CredentialScheme` to support more than one `isoNamespace`? That sure sounds like a separate issue!

Reply:
Created an issue for that: https://github.com/a-sit-plus/kmm-vc-library/issues/55